### PR TITLE
jetbrains.*: update versions

### LINKED ISSUE #292196: On new versions of RustRover (Build #RR-233.14015.153) patchelf fail on native rust helper
https://github.com/NixOS/nixpkgs/blob/16d9f67e6edb056d27c9e9f9685717f4c041a828/pkgs/applications/editors/jetbrains/default.nix#L244

This silently fail and get a hard time detecting the cause, because:
image

now seems that is statically linked:
image

### PR DESCRIPTION
**Pull request by sdht0**
## Description of changes

* Updated using `bin/update_bin.py`.
* Fixed rust-rover build, closes #292196
* Tested rust-rover, idea, goland
* Would also be nice to merge #285710 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by SebTM — APPROVED** [review]
14 packages built:
jetbrains.clion jetbrains.dataspell jetbrains.gateway jetbrains.goland jetbrains.idea-community jetbrains.idea-community-bin jetbrains.idea-ultimate jetbrains.phpstorm jetbrains.pycharm-community jetbrains.pycharm-community-bin jetbrains.pycharm-professional jetbrains.ruby-mine jetbrains.rust-rover jetbrains.webstorm

Result of `nixpkgs-review pr 293308` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-src)</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-src)</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>

Tested phpstorm locally works fine ✌🏻

**Review by niknetniko — APPROVED** [review]
Intellij Ultimate works for me.

**Comment by sdht0** [comment]
Meanwhile new version are out. Might as well update.

**Comment by lhanson** [comment]
[like]  Lyle Hanson reacted to your message:
________________________________

**Comment by sdht0** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>jetbrains.clion</li>
    <li>jetbrains.dataspell</li>
    <li>jetbrains.gateway</li>
    <li>jetbrains.goland</li>
    <li>jetbrains.idea-community (jetbrains.idea-community-src)</li>
    <li>jetbrains.idea-community-bin</li>
    <li>jetbrains.idea-ultimate</li>
    <li>jetbrains.mps</li>
    <li>jetbrains.phpstorm</li>
    <li>jetbrains.pycharm-community (jetbrains.pycharm-community-src)</li>
    <li>jetbrains.pycharm-community-bin</li>
    <li>jetbrains.pycharm-professional</li>
    <li>jetbrains.rider</li>
    <li>jetbrains.ruby-mine</li>
    <li>jetbrains.rust-rover</li>
    <li>jetbrains.webstorm</li>
  </ul>
</details>

**Review by SebTM — APPROVED** [review]

File: pkgs/applications/editors/jetbrains/default.nix [hunk 1]
```diff
@@ -240,8 +240,6 @@ rec {
           --replace-needed libssl.so.10 libssl.so \
           --replace-needed libcrypto.so.10 libcrypto.so
 
-        interp="$(cat $NIX_CC/nix-support/dynamic-linker)"
-        patchelf --set-interpreter $interp $PWD/plugins/intellij-rust/bin/linux/*/intellij-rust-native-helper
         chmod +x $PWD/plugins/intellij-rust/bin/linux/*/intellij-rust-native-helper
```
Comment (liff) [inline pkgs/applications/editors/jetbrains/default.nix:243]: Not that there’s any harm in having this, but it looks like in the latest release both the x86-64 and arm64 binaries have the executable bit set.

**Review by liff — APPROVED** [review]
Rust Rover works for me.

**Pull request merged**